src/gcp/storage.ts

```diff
@@ -229,6 +239,27 @@ export async function getBucket(bucketName: string): Promise<BucketResponse> {
   }
 }
 
+/**
+ * Gets the list of storage buckets associated with a specific project from GCP.
+ * Ref: https://cloud.google.com/storage/docs/json_api/v1/buckets/list
+ * @param {string} bucketName name of the storage bucket
+ * @return a bucket resource object
+ */
+export async function listBuckets(projectId: string): Promise<Array<string>> {
```

Comment:
note: we are going to have to deal with pagination at some point. There are other implementation for dealing with pagination in this codebase.

Reply:
I have thoughts on how dynamically paginated lists could have been implemented, it's not clear to me that `prompt` has support for a dynamic list of a select prompt. This means Victor has to query for all buckets every time before he can present UX. Most cloud products only allow 1K buckets and a bucket object is probably 1-10K, so that means the worst possible case is that we make a 10MB download. I can live with that given that this user is probably overjoyed that we even work for their project.